model/src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceModel.scala

```diff
@@ -187,11 +192,66 @@ case class EntityCopyDefinition(
                    entityNames: Seq[String]
                    )
 
+// In the next phase (GAWB-3100), this class's members will be swapped out for a singular methodUri
 case class MethodRepoMethod(
                    methodNamespace: String,
                    methodName: String,
                    methodVersion: Int
-                   )
+                   ) {
+  def asAgoraMethodUrl: String = asMethodUrlForRepo(MethodRepos.Agora.toString)
+
+  // Next phase: this method goes away
+  def asMethodUrlForRepo(repository: String): String = {
+
+    if (repository.nonEmpty && MethodRepos.withName(repository).isDefined && this.validate.isDefined)
+      s"$repository://${URLEncoder.encode(methodNamespace, "UTF-8")}/${URLEncoder.encode(methodName, "UTF-8")}/$methodVersion"
+    else
+      throw new RawlsException(
+        s"Could not generate a method URI from MethodRepoMethod with repo \'$repository\', namespace \'$methodNamespace\', name \'$methodName\', version \'$methodVersion\'"
+      )
+  }
+
+  def validate: Option[MethodRepoMethod] = {
+    if (methodNamespace.nonEmpty && methodName.nonEmpty && methodVersion > 0)
+      Some(this)
+    else
+      None
+  }
+
+}
+
+object MethodRepoMethod {
+  def apply(uri: String): MethodRepoMethod = {
+
+    (for {
+      parsedUri <- Try(parse(uri)).toOption
+      repo      <- parsedUri.scheme
+      _         <- MethodRepos.withName(repo)
+      namespace <- parsedUri.host // parser does not URL-decode host
+      parts     <- Option(parsedUri.pathParts)
+      name      <- Option(parts.head.part) // parser does URL-decode path parts
+      version   <- Try(parts(1).part.toInt).toOption
+      result    <- if (parts.size == 2) MethodRepoMethod(URLDecoder.decode(namespace, "UTF-8"), name, version).validate else None
+    } yield {
+      result
+    }).getOrElse(throw new RawlsException(s"Could not create a MethodRepoMethod from URI \'$uri\'"))
+  }
+}
+
+object MethodRepos {
+  sealed trait MethodRepository {
+    override def toString = getClass.getSimpleName.stripSuffix("$").toLowerCase
+  }
+
+  def withName(name: String): Option[MethodRepository] = name match {
+    case "agora" => Some(Agora)
+    case _ => None
+  }
+
+  case object Agora extends MethodRepository
+
+  val all: Set[MethodRepository] = Set(Agora)
```

Comment:
Consider removing unused `val`

Reply:
Generally I'd drop it, but I use it immediately in the follow-on ticket